### ISSUE TITLE
Remove extraneous edges in imported models 

### ISSUE THREAD

**Issue by suzialeksander** [geneontology/pathways2GO#276]
There are some redundant edges added to models upon import:

![image (3)](https://github.com/geneontology/pathways2GO/assets/13735608/773a0c8a-de85-4f86-8780-37549d93bb82)
Example from [YeastPathways_YEAST-SALV-PYRMID-DNTP](http://noctua-dev.berkeleybop.org/editor/graph/gomodel:YeastPathways_YEAST-SALV-PYRMID-DNTP?)

I think the other major relation that did this was "casually upstream of". 

Can we remove these before https://github.com/geneontology/pathways2GO/issues/275? 

Also tagging @thomaspd

**Comment by ukemi** [comment]
Note that this is the opposite of what we do for the Reactome models. For those, we infer and retain the causal relationship between the functions.

**Comment by dustine32** [comment]
@ukemi Right! In fact, the last commit [35f360c](https://github.com/geneontology/pathways2GO/commit/35f360cc43ffdbbaa01a0d10eacb78c57e1daf66) was made to fix a Reactome test that broke due to this. The fix ties the causal relation removal to the use of common intermediate input/output molecules as in the image from the top comment:
![image](https://github.com/geneontology/pathways2GO/assets/2678599/44f568fb-f46d-49e9-bb3f-ccce8974812f)
If relation chain 1 exists, then relation 2 will be removed.

For Reactome, (as you said above) I haven't yet switched the input/output molecule handling over to the scheme shown for relation chain 1, but weren't we [talking](https://github.com/geneontology/pathways2GO/issues/258#issuecomment-1548413465) about doing that? If yes, is the desired Reactome behavior that we have both relation chain 1 and relation 2 between input/output-connected reactions?

**Comment by ukemi** [comment]
I'd actually like both. To me removal of the causal relation (even if it can be inferred) goes against the GO-CAM data model which is supposed to link functions with causal relations..

**Comment by ukemi** [comment]
I think that the group decided that we would follow the method described above. We need to do this with the Reactome import as well.

**Comment by deustp01** [comment]
> follow the method described above

Sorry - which method above? Keep both of links 1 and 2 or only keep 1?

**Comment by dustine32** [comment]
The Reactome ticket for this is https://github.com/geneontology/pathways2GO/issues/308
